posthog/session_recordings/session_recording_api.py

```diff
@@ -16,10 +16,13 @@
 from loginas.utils import is_impersonated_session
 from rest_framework import exceptions, request, serializers, viewsets
 from rest_framework.decorators import action
+from rest_framework.renderers import JSONRenderer
```

Comment:
> If you don't use this render, but rather the DRF one, it might just handle surrogates. Thus we wouldn't need a new renderer. Did you try that?

I did... i stepped through that code to understand how it worked - so, the first version was a total copy of the `render` method.

```
        ret = json.dumps(
            data, cls=self.encoder_class,
            indent=indent, ensure_ascii=self.ensure_ascii,
            allow_nan=not self.strict, separators=separators
        )

        # We always fully escape \u2028 and \u2029 to ensure we output JSON
        # that is a strict javascript subset.
        # See: https://gist.github.com/damncabbage/623b879af56f850a6ddc
        ret = ret.replace('\u2028', '\\u2028').replace('\u2029', '\\u2029')
        return ret.encode()
```

the renderer does the above.

the error is thrown on `return ret.encode()`

the structure being rendered is valid JSON and its values sometimes contain UTF-16 strings so it's only that final encode step that fails

wrapping the string encoding in `safe_clickhouse_strings` is the smallest fix we can apply - particularly since there's no hook to affect `ret.encode` without reimplementing the method

Reply:
My guess is someone snoozed it in Sentry in the past and the change to the triggering code resurfaced it 🙈